org.eclipse.scout.rt.platform/src/main/java/org/eclipse/scout/rt/platform/html/HtmlHelper.java

```diff
@@ -272,4 +273,147 @@ else if (c == '>') {
     }
     return result.toString();
   }
+
+  /**
+   * Truncates the text content of the given html using {@link #truncate(String, Integer, boolean)} without adding an ellipsis.
+   */
+  public String truncate(String html, Integer maxLength) {
+    return truncate(html, maxLength, false);
+  }
+
+  /**
+   * Truncates the text content of the given html without removing the html tags surrounding the remaining text.
+   * The html tags that follow the remaining text and don't close a tag opened before the text may be removed.
+   *
+   * @param html
+   *     the html content to truncate
+   * @param maxLength
+   *     the maximum length the truncated text will have
+   * @param addEllipsis
+   *     true to add an ellipsis character at the position where the text was truncated
+   */
+  public String truncate(String html, Integer maxLength, boolean addEllipsis) {
+    if (StringUtility.isNullOrEmpty(html)) {
+      return html;
+    }
+    if (maxLength <= 0) {
+      return addEllipsis ? "…" : "";
+    }
+    int size = html.length();
+    if (size < maxLength) {
+      // Content fits completely with all tags, just return it as it is
+      return html;
+    }
+
+    boolean inTag = false;
+    boolean inEntity = false;
+    boolean inAttr = false;
+    boolean inComment = false;
+    char attrChar = '"';
+    int numOpenTags = 0;
+    int numChars = 0;
```

Comment:
```suggestion
    int numTextChars = 0;
```
Avoid confusion with the "chars" we loop over.